Subject: RT: Make cpupri use a raw-spinlock for -rt
From: Gregory Haskins <ghaskins@novell.com>

Scheduler related locks must be raw since we can reschedule in interrupt ctx

Signed-off-by: Gregory Haskins <ghaskins@novell.com>
Index: linux-2.6.23/kernel/sched_cpupri.h
===================================================================
--- linux-2.6.23.orig/kernel/sched_cpupri.h
+++ linux-2.6.23/kernel/sched_cpupri.h
@@ -13,8 +13,8 @@
 
 struct cpupri_vec
 {
-	spinlock_t lock;
-	cpumask_t  mask;
+	raw_spinlock_t lock;
+	cpumask_t      mask;
 };
 
 struct cpupri {
